tests/python/gaia-ui-tests/gaiatest/apps/camera/app.py

```diff
@@ -87,7 +87,7 @@ def tap_select_button(self):
         self.tap_element_from_system_app(select)
 
         # Fall back to app beneath the picker
-        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name != self.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url != self.manifest_url)
```

Comment:
I think this should be self.wait_to_be_displayed().
But you also need to add the manifest_url property for this app object, no?